### ISSUE TITLE
[BUG] Session is blocked in Russia?

### ISSUE THREAD

**Issue by keybreak** [oxen-io/session-android#1617]
### Code of conduct

- [X] I have read and agree to adhere to the [Code of Conduct](https://github.com/oxen-io/session-android/blob/master/CODE_OF_CONDUCT.md)

### Self-training on how to write a bug report

- [X] I have learned [how to write a bug report](https://www.browserstack.com/guide/how-to-write-a-bug-report)

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have a friend living in Russia, he's not very tech (GitHub) savvy, so i'd have to be a mediator for him.

Since yesterday both Android and Linux Session can not connect to network, path is always red and rotating.
And, unfortunately, looks like it's censorship ministry of Russia started blocking Session, just like recently they did with Signal (btw it's censorship circumvention toggle seems to work against it), because when running through VPN / Orbot on Android he says it can connect.

That is a big problem, since Session / lokinet network suppose to circumvent censorship and bring freedom against this kind of tyranny.

We need some sort of working circumvention mechanism ASAP, i suggest looking into something like Tor's webtunnel, as it seems to work perfectly so far inside China, Russia and Iran against Tor blockade.

### Session Version

1.19.1 (3795)

**Comment by keybreak** [comment]
Here's Russian forum that specialises on censorship circumvention, hopefully their findings will help to debug it further:
https://ntc.party/t/%D0%B1%D0%BB%D0%BE%D0%BA%D0%B8%D1%80%D0%BE%D0%B2%D0%BA%D0%B0-%D0%BC%D0%B5%D1%81%D1%81%D0%B5%D0%BD%D0%B4%D0%B6%D0%B5%D1%80%D0%B0-session/9223

Use some russian-english translator.

**Comment by 1e100** [comment]
Yeah, I recommended it to my friends in Russia after the founder of Telegram got in jail in France (with a soldering iron up his rectum, no doubt), and was surprised to discover that this "anti censorship" messenger is fully nuked from the orbit in Russia, probably using the same DPI filter as Signal.

**Comment by keybreak** [comment]
> Yeah, I recommended it to my friends in Russia

Same, i did so long ago for anyone who was wise and willing to listen, because of war intensification, knowing full well it will be extremely hard for any free speech in there....was very disappointed to find out Session was so easily blocked by IP.

@KeeJef
That should be priority number 1 right now, because otherwise Session reputation will fade very quickly...well it already does, and it's sad to see :frowning: 

No amount of twitter / youtube PR / marketing would help to mitigate blockade of people's friends and relatives, you know. :rofl:

**Comment by 1e100** [comment]
Yeah for that matter after TikTok gets blocked here in the US it's only a matter of time before the government tries to ban everything it can't monitor and surveil. So while it's not an immediate need (for the US and other increasingly autocratic Western countries such as the UK, Australia, Germany and France), I think implementing extreme censorship resistance would be a worthwhile goal in the long run.

**Comment by KeeJef** [comment]
I'm aware of the blocking of Session that is ongoing in Russia and other countries. Session has plans to make make the app censorship-resistant out of the box moving forward to resolve these issues. It's a challenging problem to solve, especially when censorship occurs at the Service Node level, as Service Node IPs need to be advertised to Session clients for them to connect to the network and store or route messages. Censors can gather and block all Service Node IP addresses to prevent access to the network. This generally cant be resolved by adding DPI protections or changing the traffic signature of the application, as its the actual IPs of the network being blocked.

There are several approaches Session is considering to address this issue, including:

- Adding support for clients to define and connect to local or remote proxies.
- Allowing clients to define and connect to their own Seed Nodes.
- Client support for domain fronting and other censorship evasion techniques.

In saying this the current focus of the Session team is on the following issues:

1. Reducing crashes.
2. Fixing issues related to sending messages and attachments.
3. Reducing ANRs (App Not Responding) and freezes.
4. Overhauling closed groups implementation and unifying strings across platforms.

The Session team believes these areas require the most immediate attention. If existing, non censored users can't use the app in a functional way, Session will have trouble growing. Once these issues are resolved, the team can start focusing on expanding the reach of Session to users in more censored countries.

**Comment by keybreak** [comment]
Thx for finally acknowledging the problem, it's much appreciated :+1: 

> Client support for domain fronting and other censorship evasion techniques.

Sounds like a good long-term plan, Tor's webtunnel style.

**Comment by keybreak** [comment]
> Probably one of the most effective & straightforward solution to the problem would be to add the ability to use a Tor-bridge on the first hop?

Definitely not, because Tor is blocked in Russia too (without bridges)...so it would just add more complexity and another dependency that is complex enough.

It's best to do it all in-house, since lokinet is also onion-network anyway, Session should straighten it directly.
